Octokit/Clients/TeamsClient.cs

```diff
@@ -37,6 +38,28 @@ public Task<Team> Get(int id)
             return ApiConnection.Get<Team>(endpoint, null, AcceptHeaders.NestedTeamsPreview);
         }
 
+        /// <summary>
+        /// Gets a single <see cref="Team"/> by name.
+        /// </summary>
+        /// <remarks>
+        /// this is more expensive than using the team ID, as first we
+        /// perform a lookup to determine matching teams.
+        /// </remarks>
+        /// <param name="orgName">The organization name.</param>
+        /// <param name="teamName">The team name.</param>
+        /// <exception cref="ArgumentNullException">if the list of teams returned from GetAll is null.</exception>
+        /// <exception cref="InvalidOperationException">if no matching team name is found.</exception>
+        /// <returns>The <see cref="Team"/> with the given identifier.</returns>
+        public async Task<Team> Get(string orgName, string teamName)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(orgName, nameof(orgName));
+            Ensure.ArgumentNotNullOrEmptyString(teamName, nameof(teamName));
+
+            var allTeams = await this.GetAll(orgName);
+
+            return allTeams.First(x => string.Equals(x.Name, teamName, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
I'm not a fan of `.First` here because it's behaviour is different to how Octokit itself handles cases where an entity cannot be found. In these cases the API response returns a 404, which we convert into a `NotFoundException` and throw as part of the plumbing. I'd prefer we stick to this pattern to avoid callers having to handle multiple things.